ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/services/FragmentationServiceImpl.java

```diff
@@ -24,58 +19,42 @@ public class FragmentationServiceImpl implements FragmentationService {
     private final LdesMemberRepository ldesMemberRepository;
     private final LdesFragmentRespository ldesFragmentRespository;
 
+    private final FragmentCreator fragmentCreator;
+
     public FragmentationServiceImpl(LdesConfig ldesConfig, ViewConfig viewConfig, LdesMemberRepository ldesMemberRepository,
-                                    LdesFragmentRespository ldesFragmentRespository) {
+                                    LdesFragmentRespository ldesFragmentRespository, FragmentCreator fragmentCreator) {
         this.ldesConfig = ldesConfig;
         this.viewConfig = viewConfig;
         this.ldesMemberRepository = ldesMemberRepository;
         this.ldesFragmentRespository = ldesFragmentRespository;
+        this.fragmentCreator = fragmentCreator;
     }
 
     @Override
     public LdesMember addMember(LdesMember ldesMember) {
-        ldesMember.resetLdesMemberView(ldesConfig);
-        LdesFragment ldesFragment = ldesFragmentRespository.retrieveLastFragment(ldesConfig.getCollectionName())
-                .map(fragment -> {
-                    if(fragment.getMembers().size() >= fragment.getFragmentInfo().getMemberLimit()) {
-                        return makeNewFollowingFragment(fragment, ldesMember, "tree:GreaterThanRelation", "tree:LesserThanRelation");
-                    }
-                    else {
-                        return fragment;
-                    }
-                })
-                .orElse(makeNewFragment(ldesMember));
-        addLdesMembertoLdesFragment(ldesMember, ldesFragment);
+        ldesMember.replaceTreeMemberStatement(ldesConfig.getHostName(), ldesConfig.getCollectionName());
+        LdesFragment ldesFragment = retrieveLastFragmentOrCreateNewFragment(ldesMember);
+        ldesFragment.addMember(ldesMember);
```

Comment:
This seems to not take into account if you have to shift members in a fragment (to a new fragment)